proc/cond.c

```diff
@@ -95,7 +125,44 @@ int proc_condWait(int c, int m, time_t timeout)
 		return -EINVAL;
 	}
 
-	err = proc_lockWait(&cond->queue, &mutex->lock, timeout);
+	if (timeout != 0) {
+		switch (cond->attr.clock) {
+			case COND_CLOCK_REALTIME:
+				proc_gettime(&abstime, &offs);
+				if (abstime + offs > timeout) {
+					err = -ETIME;
+					break;
+				}
+
+				abstime = timeout - offs;
+				break;
+
+			case COND_CLOCK_MONOTONIC:
+				proc_gettime(&abstime, NULL);
+				if (abstime > timeout) {
+					err = -ETIME;
+					break;
+				}
+
+				abstime = timeout;
+				break;
+
+			case COND_CLOCK_RELATIVE:
+				/* and COND_CLOCK_DEFAULT */
+				proc_gettime(&abstime, NULL);
+				abstime += timeout;
+				break;
+
+			default:
+				/* Wrong clock type, should never happen */
+				err = -EINVAL;
+				break;
+		}
+	}
```

Comment:
This time getter might be a good candidate for a general function in proc/threads module - it might be useful for sleep functions too